File: Assignment20/Q4.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q4.java
//      Project   : Logic Building Assignment 20
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : LargestDigitLogic
//      Method Name: findLargestDigit
//      Input      : Integer
//      Output     : Void
//      Description: Find the largest digit in
//                   the given number
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class LargestDigitLogic
{
    void findLargestDigit(int num)
    {
        int digit = 0;          // Store extracted digit
        int iMax = 0;           // Store largest digit

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Extract digits and find largest digit
        while(num != 0)
        {
            digit = num % 10;

            if(digit > iMax)
            {
                iMax = digit;
            }

            num = num / 10;
        }

        // Display result
        System.out.println("Largest digit is : " + iMax);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q4
{
    public static void main(String args[])
    {
        // Create object of LargestDigitLogic class
        LargestDigitLogic obj = new LargestDigitLogic();

        // Function call
        obj.findLargestDigit(83429);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 83429
//
//      Output :
//      Largest digit is : 9
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
////////////////////////////////////////////////////////////////////

File: Assignment23/Q4.c
///////////////////////////////////////////////////////////////////
//
//      File Name : Q4.c
//      Project   : Logic Building Pattern Assignment
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Including Header File
//
///////////////////////////////////////////////////////////////////

#include<stdio.h>

///////////////////////////////////////////////////////////////////
//
//      Function Name : Pattern
//      Input         : Integer, Integer
//      Output        : Void
//      Description   : Display '*' at odd columns
//                      and '#' at even columns for
//                      every row
//      Date          : 08/06/2026
//      Author        : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

void Pattern(int iRow, int iCol)
{
    int i = 0;      // Row counter
    int j = 0;      // Column counter

    // Traverse each row
    for(i = 1; i <= iRow; i++)
    {
        // Traverse each column
        for(j = 1; j <= iCol; j++)
        {
            // Print '*' at odd position
            if(j % 2 != 0)
            {
                printf("* ");
            }
            // Print '#' at even position
            else
            {
                printf("# ");
            }
        }

        // Move cursor to next line
        printf("\n");
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Function
//
///////////////////////////////////////////////////////////////////

int main()
{
    int iValue1 = 0;      // Store number of rows
    int iValue2 = 0;      // Store number of columns

    // Accept rows and columns from user
    printf("Enter number of rows and columns : ");
    scanf("%d %d", &iValue1, &iValue2);

    // Function call
    Pattern(iValue1, iValue2);

    return 0;
}

///////////////////////////////////////////////////////////////////
//
//      Input  :
//      Rows = 3
//      Columns = 4
//
//      Output :
//      * # * #
//      * # * #
//      * # * #
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(iRow * iCol)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////

